_examples/01_one_account_one_region/README.md

```diff
@@ -0,0 +1,155 @@
+│   |   └── terraform.tfvars        <- component configuration
+│   └── vpc
+│       └── terraform.tfvars
+└── terraform.tfvars                <- root configuration
+```
+
+The `live/` directory is the root of our Terragrunt configuration hierarchy.
+It contains a single `terraform.tfvars` file, which we will call
+the **root configuration**. The root configuration is optional, but allows 
+us to define general configuration that applies to all components in our infrastructure
+in a single place.
+
+The root contains a child directory for each environment in the infrastructure
+(e.g. production, staging), which we will call **environment directories**.
+The directory structure shown above contains two environment directories:
+`production` and `staging`.
+Environment directories do not normally contain any files. Instead, they are
+the location where Terragrunt commands such as `plan-all` and `apply-all`
+are executed to instantiate all of the infrastructure components for the
+given environment.
+
+Environment directories contain a child directory for each component of 
+the infrastructure, which we will call **component directories**.
+For example, a component might be a single app, a single MySQL DB, or a single VPC.
+The directory structure shown above contains 4 component directories: `frontend-app` and
+`vpc` in each environment.
+
+Each component directory contains a single `terraform.tfvars` file, which
+defines the configuration for that specific component. We will call this
+`terraform.tfvars` file the **component configuration**. The directory
+structure shown above contains 4 component configurations: one for
+each component in each environment.
+In a component configuration, we can use the `include` stanza to instruct 
+Terragrunt to locate and copy configuration from the root configuration file. 
+This allows us to keep our configurations DRY. Review the example
+code in this repo to see how this works in practice.
+
+## Each infrastructure component is defined in a Terraform module
+In this example, our infrastructure contains two components: a VPC and a frontend_app.
+Each of these components is defined in its own Terraform module:
+[conorgil/terragrunt_sample_vpc](https://github.com/conorgil/terragrunt_sample_vpc)
+and
+[conorgil/terragrunt_sample_frontend_app](https://github.com/conorgil/terragrunt_sample_frontend_app)
+respectively.
+
+There are a few important things to note about the Terraform modules:
+1. The code in each of these repos is 100% regular plain old Terraform code. These
+   Terraform modules can be used, without modification, in any project that does
+   not use Terragrunt and instead uses Terarform directly.
+1. Each module defines a variable for any value that might change between invocations
+   of the module. For example, values that might change between production and staging
+   environments.
+1. Each module enables a backend for storing remote state. The backend intentionally

```

Comment:
Does this now impose the requirement of remote state inside the module? IE what if someone has modules they don't want to use remote state for? I can't help but feel like this sort of decision belongs in the in the component configuration, not the module.